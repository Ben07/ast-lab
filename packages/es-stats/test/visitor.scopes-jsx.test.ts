import fs from 'fs';
import path from 'path';
import { parse } from '@babel/parser';
import traverse from '@babel/traverse';
import debug from 'debug';
import { Declarations } from 'ast-lab-types';
import createVisitor from '../src/visitors/rootRelation';

const dirs = ['jsx'];
describe('rootRelation visitors', () => {
  dirs.forEach(dir => {
    const dirPath = path.resolve(__dirname, '__fixtures__', dir);
    const files = fs.readdirSync(dirPath);
    files.forEach(file => {
      test(`${dir}/${file}`, () => {
        debug('es-stats:test')(`${dir}/${file}`);
        let res = {} as Declarations;
        const code = fs.readFileSync(path.resolve(dirPath, file), 'utf-8');
        traverse(
          // @ts-ignore
          parse(code, {
            sourceType: 'module',
            plugins: ['classProperties', 'jsx'],
          }),
          createVisitor(res)
        );
        expect(res).toMatchSnapshot();
      });
    });
  });
});
