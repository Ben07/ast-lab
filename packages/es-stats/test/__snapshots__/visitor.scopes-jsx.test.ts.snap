// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rootRelation visitors jsx/functionComponent.js 1`] = `
Object {
  "ChildFunctionComponentHasProps1": Object {
    "dependencies": Array [
      "ParentFunctionComponent",
    ],
    "loc": SourceLocation {
      "end": Position {
        "column": 1,
        "line": 13,
      },
      "start": Position {
        "column": 7,
        "line": 9,
      },
    },
  },
  "ChildFunctionComponentHasProps2": Object {
    "dependencies": Array [
      "ParentFunctionComponent",
    ],
    "loc": SourceLocation {
      "end": Position {
        "column": 1,
        "line": 21,
      },
      "start": Position {
        "column": 7,
        "line": 15,
      },
    },
  },
  "ChildFunctionComponentJSXMemberExpression": Object {
    "dependencies": Array [
      "ParentFunctionComponentObj",
    ],
    "loc": SourceLocation {
      "end": Position {
        "column": 1,
        "line": 30,
      },
      "start": Position {
        "column": 7,
        "line": 28,
      },
    },
  },
  "ChildFunctionComponentNoProps": Object {
    "dependencies": Array [
      "ParentFunctionComponent",
    ],
    "loc": SourceLocation {
      "end": Position {
        "column": 1,
        "line": 7,
      },
      "start": Position {
        "column": 7,
        "line": 5,
      },
    },
  },
  "ParentFunctionComponent": Object {
    "dependencies": Array [
      "div",
      "JSON",
    ],
    "loc": SourceLocation {
      "end": Position {
        "column": 1,
        "line": 3,
      },
      "start": Position {
        "column": 0,
        "line": 1,
      },
    },
  },
  "ParentFunctionComponentObj": Object {
    "dependencies": Array [
      "ParentFunctionComponent",
    ],
    "loc": SourceLocation {
      "end": Position {
        "column": 2,
        "line": 26,
      },
      "start": Position {
        "column": 0,
        "line": 24,
      },
    },
  },
}
`;
